### PR TITLE
Add some more tests for legacy data marshalling

### PR DESCRIPTION
**Pull request by RustyHMCTS**
**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[*] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/opal/dto/AccountSearchDto.java [hunk 1]
```diff
@@ -29,4 +30,11 @@ public class AccountSearchDto implements ToJsonString {
     private String majorCreditor;
     /** Unsure. */
     private String tillNumber;
+
+    public Long getNumericCourt() {
```
Comment (sabahirfan) [inline src/main/java/uk/gov/hmcts/opal/dto/AccountSearchDto.java:34]: How about  if we return `Optional<Long>` instead returning `null` t line 38 ?

Comment (RustyHMCTS) [reply src/main/java/uk/gov/hmcts/opal/dto/AccountSearchDto.java:34]: I did think about doing that, but in the places where this is being utilised, I actually want a null.

File: src/main/java/uk/gov/hmcts/opal/dto/legacy/DefendantAccountSearchCriteria.java [hunk 2]
```diff
@@ -16,37 +17,50 @@
 @AllArgsConstructor
 public class DefendantAccountSearchCriteria implements ToJsonString {
 
+    @JsonInclude(JsonInclude.Include.NON_NULL)
     @JsonProperty("account_number")
     private String accountNumber;
 
+    @JsonInclude(JsonInclude.Include.NON_NULL)
```
Comment (sabahirfan) [inline src/main/java/uk/gov/hmcts/opal/dto/legacy/DefendantAccountSearchCriteria.java:24]: Is there an option to configure this on class level for all the fields?

Comment (RustyHMCTS) [reply src/main/java/uk/gov/hmcts/opal/dto/legacy/DefendantAccountSearchCriteria.java:24]: Yes - have done that in new commit

File: src/test/java/uk/gov/hmcts/opal/dto/legacy/DefendantAccountSearchCriteriaTest.java [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static uk.gov.hmcts.opal.dto.legacy.DefendantAccountSearchCriteria.DefendantAccountSearchCriteriaBuilder;
+
+@Slf4j
+public class DefendantAccountSearchCriteriaTest {
+
+    @Test
+    public void testBuilder() throws InterruptedException {
```
Comment (sabahirfan) [inline src/test/java/uk/gov/hmcts/opal/dto/legacy/DefendantAccountSearchCriteriaTest.java:17]: In my opinion we don't need to write unit tests for the generated code.

**Pull request merged**